String.cpp

```diff
@@ -0,0 +1,126 @@
+#include "String.h"
+
+String::String() {
+    str = new char[1];
+    str[0] = '\0';
+    size = 0;
+}
+
+String::String(const char *str) {
+    size = strlen(str);
+    this->str = new char[size + 1];
+    strcpy(this->str, str);
+}
+
+void String::copyStr(const String &str) {
+    size = str.size;
+    this->str = new char[size + 1];
+    strcpy(this->str, str.str);
+}
+
+void String::clear() {
+    delete[] str;
+    size = 0;
+}
+
+String::~String() {
+    clear();
+}
+
+String::String(const String &str) {
+    copyStr(str);
+}
+
+String& String::operator=(const String &str) {
+    if (this != &str) {
+        clear();
+        copyStr(str);
+    }
+    return *this;
+}
+
+char& String::operator[](const unsigned int pos) {
+    return str[pos];
+}
+
+char String::operator[](const unsigned int pos) const {
+    return str[pos];
+}
+
+String operator+(const String& str1, const String& str2) {
+    String res;
+    res.clear();
+    res.str = new char[str1.size + str2.size + 1];
+    strcpy(res.str, str1.str);
+    strcat(res.str, str2.str);
+    res.size = str1.size + str2.size;
+    return res;

```

Comment:
create consturctor that preallocates memory. Do not use clear func outside copy constructor and oper=.